pkg/validator/cluster.go

```diff
@@ -23,18 +23,31 @@ import (
 
 	kdv1 "github.com/bluek8s/kubedirector/pkg/apis/kubedirector.bluedata.io/v1alpha1"
 	"github.com/bluek8s/kubedirector/pkg/catalog"
+	"github.com/bluek8s/kubedirector/pkg/observer"
 	"github.com/bluek8s/kubedirector/pkg/reconciler"
 	"github.com/bluek8s/kubedirector/pkg/shared"
 	"k8s.io/api/admission/v1beta1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // membersPatchSpec is used to create the PATCH operation for adding a default
-// member count to a role.
+// member count to a role and setting up default storageClassName
 type membersPatchSpec struct {
-	Op    string `json:"op"`
-	Path  string `json:"path"`
-	Value int32  `json:"value"`
+	Op    string            `json:"op"`
+	Path  string            `json:"path"`
+	Value membersPatchValue `json:"value"`
+}
+
+type membersPatchValue struct {
+	ValueInt *int32
+	ValueStr *string
+}
```

Comment:
I'm pleasantly surprised that this sort of thing works as a patch value. Where did you see/find this approach? We could use it to also simplify stuff like the patches in executor/cluster.go.

Reply:
Ohhhh I see the custom marshalling below. Nice.